### PR TITLE
Move all singletons to new mongo collection

### LINKED ISSUE #291: Move all singletons to a new Mongo collection
Move the following singleton documents to a new Mongo collection called `singletons`.
- config
- version
- gears

Ultimately, gears should be migrated to a separate collection.

### PR DESCRIPTION
**Pull request by nagem**
Closes #291

### CODE REVIEW

File: api/config.py [hunk 1]
```diff
@@ -52,27 +52,28 @@
     },
 }
 
-__config = copy.deepcopy(DEFAULT_CONFIG)
+def apply_env_variables(config):
+    # Overwrite default config values with SCITRAN env variables if available
+    for outer_key, scoped_config in config.iteritems():
+        if outer_key in ['core', 'site', 'auth', 'persistent']:
```
Comment (gsfr) [inline api/config.py]: What's the purpose of this check? I wouldn't anticipate a code change when adding a new outer config key.

**Comment by gsfr** [comment]
LGTM, pending one question. Also, untested. And I'm still sick and probably shouldn't be trusted with code reviews today.

File: api/config.py (hunk 1 above)
Comment (nagem) [inline api/config.py]: It's because this can now be called on a db config object, which has keys like `created` and `modified`.

Comment (gsfr) [inline api/config.py]: Hmm, I don't like repeating the keys. Does this really have to be a callable? You're only calling it once, and that's right below the definition.

Comment (nagem) [inline api/config.py]: I also call it [here](https://github.com/scitran/core/pull/295/files/a935657d5a852fc569c6394c2ae0c7de3e7e71a7#diff-283256b6a6d2885d51431bd002292cdaR199) to preserve any environment variables on startup after applying configs from the db. I could also check if `scoped_config` has an `__iter__` property so we don't try to iterate over the dates.

Comment (nagem) [inline api/config.py]: Before we ignored anything in db config object on startup and overwrote it with env vars and default values. This change is setting the order of precedence to be env > db > default on startup.

Comment (gsfr) [inline api/config.py]: I didn't realize that. Good catch. You could `try` the iteration or use `DEFAULT_CONFIG.keys()`.

**Pull request merged**